File: ios-app/BayitPlusApp/Models/RadioModels.swift
import Foundation

// MARK: - Radio

/// Response from GET /api/v1/radio/stations
struct StationsResponse: Decodable, Sendable {
    let stations: [RadioStationItem]
    let total: Int
}

/// A radio station
struct RadioStationItem: Decodable, Sendable, Identifiable {
    let id: String
    let name: String?
    let description: String?
    let logo: String?
    let genre: String?
    let cultureId: String?
    let currentShow: String?
    let currentSong: String?
}

/// Response from GET /api/v1/radio/{station_id}
struct RadioStationDetail: Decodable, Sendable, Identifiable {
    let id: String
    let name: String?
    let description: String?
    let logo: String?
    let genre: String?
    let currentShow: String?
    let currentSong: String?
}

/// Response from GET /api/v1/radio/{station_id}/stream
struct RadioStreamResponse: Decodable, Sendable {
    let url: String
    let type: String
}

File: android-app/core/core-model/src/main/java/tv/bayit/plus/core/model/RadioModels.kt
package tv.bayit.plus.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Response from GET /api/v1/radio/stations */
@Serializable
data class StationsResponse(
    val stations: List<RadioStationItem> = emptyList(),
    val total: Int,
)

/** A radio station. */
@Serializable
data class RadioStationItem(
    val id: String,
    val name: String? = null,
    val description: String? = null,
    val logo: String? = null,
    val genre: String? = null,
    @SerialName("culture_id") val cultureId: String? = null,
    @SerialName("current_show") val currentShow: String? = null,
    @SerialName("current_song") val currentSong: String? = null,
)

/** Response from GET /api/v1/radio/{station_id} */
@Serializable
data class RadioStationDetail(
    val id: String,
    val name: String? = null,
    val description: String? = null,
    val logo: String? = null,
    val genre: String? = null,
    @SerialName("current_show") val currentShow: String? = null,
    @SerialName("current_song") val currentSong: String? = null,
)

/** Response from GET /api/v1/radio/{station_id}/stream */
@Serializable
data class RadioStreamResponse(
    val url: String,
    val type: String,
)
